python-backend/app/api/users/minespace/resources/minespace_user.py

```diff
@@ -0,0 +1,53 @@
+import uuid
+
+from flask_restplus import Resource, reqparse
+from flask import request
+from app.extensions import api
+from ....utils.access_decorators import requires_role_mine_admin
+from ....utils.resources_mixins import UserMixin, ErrorMixin
+from ..models.minespace_user import MinespaceUser
+from ..models.minespace_user_mine import MinespaceUserMine
+from app.extensions import db
+
+
+class MinespaceUserResource(Resource, UserMixin, ErrorMixin):
+    parser = reqparse.RequestParser()
+    parser.add_argument('email', type=str, location='json', required=True)
+    parser.add_argument('mine_guids', type=list, location='json', required=True)
+
+    @api.doc(params={'user_id': 'User id.'})
+    @requires_role_mine_admin
+    def get(self, user_id=None):
+        if user_id or request.args.get('email'):
+            user = MinespaceUser.find_by_id(user_id)
+            if not user:
+                user = MinespaceUser.find_by_email(request.args.get('email'))
+                if not user:
+                    return self.create_error_payload(404, "user not found"), 404
+            return user.json()
+        users = MinespaceUser.query.filter_by(deleted_ind=False).all()
+        return {'users': [x.json() for x in users]}
+
+    @api.doc(params={'user_id': 'User id.'})
+    @requires_role_mine_admin
+    def post(self, user_id=None):
+        if user_id:
+            return self.create_error_payload(400, "unexpected user guid"), 400
```

Comment:
Is there a benefit to this check compared to the default invalid route error that I thiiink would occur here?

Reply:
`api.add_resource(MinespaceUserResource, '/minespace', '/minespace/<user_id>')` both of these urls send to this resource, but we need make sure for post they only use the first one, sending a POST with a guid is invalid. editing an existing record should be a PUT.

Reply:
Weird. But it looks like this is consistent with our existing patterns, so I'll mark these comments as resolved